pkg/run/run.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Entrypoint to run the app
+
+// Package run provides a function that can be invoked inside of main to set up
+// all the service-standard components we expect every service to run.
+//
+// clients should provide any runners they require as part of the app to the
+// Run function with OptAddRunner, and then call `Run` in their main function.
```

Comment:
suggestion: it might be worth adding an example of how to use this.

Reply:
Added a testable example... should probably add a note that it exists here.